src/client/common/application/commands.ts

```diff
@@ -179,7 +179,7 @@ export interface ICommandNameArgumentTypeMapping extends ICommandNameWithoutArgu
     [DSCommands.ShowDataViewer]: [IShowDataViewerFromVariablePanel];
     [DSCommands.RefreshDataViewer]: [];
     [DSCommands.ClearSavedJupyterUris]: [];
-    [DSCommands.SelectJupyterURI]: [undefined, 'toolbar' | 'nativeNotebookStatusBar' | undefined];
+    [DSCommands.SelectJupyterURI]: [boolean | undefined, string | undefined, 'toolbar' | 'nativeNotebookStatusBar' | undefined];
```

Comment:
This actually won't work for the regular command I think. Other instances of calling this command must already pass the string as the second argument.

You could do this by making the second parameter a ```URI  | 'toolbar' | 'nativeNotebookStatusBar'``` and then check if the parameter is a URI in your handler.

Reply:
True, it seems to work in the default case because its called without anything, but if there are instances its been used I suppose there is no reason to break that compatibility. Will push up a change with it allowing a `Uri` object and use `if instanceof Uri`,